osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/factories/DlmsConnector.java

```diff
@@ -20,6 +39,27 @@
 public abstract class DlmsConnector {
   private static final Logger LOGGER = LoggerFactory.getLogger(DlmsConnector.class);
 
+  private static final Map<String, FunctionalExceptionType> errorMap = new HashMap<>();
+
+  static {
+    errorMap.put("AUTHENTICATION_ERROR", AUTHENTICATION_ERROR);
+    errorMap.put("AUTHENTICATION_REQUIRED", AUTHENTICATION_REQUIRED);
+    errorMap.put("Connection refused", CONNECTION_REFUSED);
+    errorMap.put("Connection reset", CONNECTION_RESET);
+    errorMap.put("Connection timed out", CONNECTION_TIMED_OUT);
+    errorMap.put("CONNECTION_ESTABLISH_ERROR", CONNECTION_ESTABLISH_ERROR);
+    errorMap.put("Socket was closed by remote host.", SOCKET_CLOSED_BY_REMOTE);
+    errorMap.put("Unable to decypher/decrypt xDLMS pdu", UNABLE_TO_DECYPHER);
+    errorMap.put("WRAPPER_HEADER_INVALID_VERSION", WRAPPER_HEADER_ERROR);
+    errorMap.put("WRAPPER_HEADER_INVALID_SRC_DEST_ADDR", WRAPPER_HEADER_ERROR);
+    errorMap.put("WRAPPER_HEADER_INVALID_PAYLOAD_LENGTH", WRAPPER_HEADER_ERROR);
+    errorMap.put("WRAPPER_HEADER_INVALID", WRAPPER_HEADER_ERROR);
+    errorMap.put("ILLEGAL_RESPONSE", ILLEGAL_RESPONSE);
+    errorMap.put("RESPONSE_TIMEOUT", RESPONSE_TIMEOUT);
+    errorMap.put("UNKNOWN_ASSOCIATION_RESULT", UNKNOWN_ASSOCIATION_RESULT);
+    errorMap.put("UNKNOWN", UNKNOWN);
```

Comment:
Can we rely on a fixed order of the entryset in the stream below?

Reply:
Changed to a SortedMap